docs/user/openstack/deploy_sriov_workers.md

```diff
@@ -0,0 +1,318 @@
+
+```sh
+oc get machineset -n openshift-machine-api <machineset_name> -o yaml > sriov_machineset.yaml
+```
+
+When editing an existing machineSet, or a copy of one, to create SR-IOV compute nodes, add each subnet configured for SR-IOV to the machineSet's `providerSpec`. The example below will attach ports from the `radio` and `uplink` subnets, from the previous example, to all the compute nodes in the machineSet. The `nicType` needs to be set to `direct` for all subnets that are using SR-IOV. The number of ports to attach for a given subnet or network is specified by setting `portCount`. For the sake of example, we attach four ports for each SR-IOV subnet on every machine. The tags added to the `portTags` list will be added to any port attached to machines from a given subnet or network. In this example, all ports attached to machines from the `radio` subnet will get the "sriov" and "radio" tags added to them, and all ports attached to machiens from the `uplink` subnet will get the "sriov" and "uplink" tags added to them.
+
+```yaml
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  labels:
+    machine.openshift.io/cluster-api-cluster: <infrastructure_ID>
+    machine.openshift.io/cluster-api-machine-role: <node_role>
+    machine.openshift.io/cluster-api-machine-type: <node_role>
+  name: <infrastructure_ID>-<node_role>
+  namespace: openshift-machine-api
+spec:
+  replicas: <number_of_replicas>
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-cluster: <infrastructure_ID>
+      machine.openshift.io/cluster-api-machineset: <infrastructure_ID>-<node_role>
+  template:
+    metadata:
+      labels:
+        machine.openshift.io/cluster-api-cluster: <infrastructure_ID>
+        machine.openshift.io/cluster-api-machine-role: <node_role>
+        machine.openshift.io/cluster-api-machine-type: <node_role>
+        machine.openshift.io/cluster-api-machineset: <infrastructure_ID>-<node_role>
+    spec:
+      providerSpec:
+        value:
+          apiVersion: openstackproviderconfig.openshift.io/v1alpha1
+          cloudName: openstack
+          cloudsSecret:
+            name: openstack-cloud-credentials
+            namespace: openshift-machine-api
+          flavor: <nova_flavor>
+          image: <glance_image_name_or_location>
+          serverGroupID: <optional_UUID_of_server_group>
+          kind: OpenstackProviderSpec
+          networks:
+          - subnets:
+            - id: <machines_subnet_uuid>
+            - filter:
+                name: radio
+              vnicType: direct
+              portCount: 4
+              portTags:
+                - "sriov"
+                - "radio"
+            - filter:
```

Comment:
I think another network needs to be defined here with network UUID, and its own subet section?

Reply:
I think we have the two from the example covered: radio, and uplink